src/core/namespace.c

```diff
@@ -2157,7 +2157,10 @@ int setup_namespace(
                                 goto finish;
                 }
 
-                if (ns_info->protect_hostname) {
+                /* If proc is mounted with subset=pid then neither of the two
+                 * paths will exist, i.e. they are implicitly protected by the
+                 * mount option. */
+                if (ns_info->protect_hostname && !ns_info->proc_subset == PROC_SUBSET_PID) {
```

Comment:
Maybe use `.ignore = ns_info->proc_subset == PROC_SUBSET_PID` instead? Using proc_subnet_pid can fail (eg, old kernel), in which case this should still apply I think

Reply:
Ha, we had the same worry. :)
Changed it to set `.ignore` in case `PROC_SUBSET_PID` is requested.